lib/src/components/NxDonutChart/NxDonutChart.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FunctionComponent, ReactElement } from 'react';
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+const NxDonutChart: FunctionComponent<Props> =
+  function NxDonutChart(props): ReactElement<Props> {
+    const decimal = props.percent / 100;
+    const diameter = props.outerRadius * 2;
+    const strokeWidth = props.outerRadius - props.innerRadius;
+    const r = props.innerRadius + strokeWidth / 2;
+    const circumference = 2 * Math.PI * r;
+    const strokeDasharray = [decimal * circumference, (1.0 - decimal) * circumference];
```

Comment:
I've discovered that there's a [`pathLength`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/pathLength) attribute that allows us to define custom sizing semantics for things like the dasharray.  That is, if you set `pathLength="100"`, you can then just set the dasharray using the actual percentage followed by 100 - dashoffset notwithstanding that is, but see one of the other threads on my idea to avoid that confusing attribute.

Reply:
> I've discovered that there's a [`pathLength`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/pathLength) attribute that allows us to define custom sizing semantics for things like the dasharray. That is, if you set `pathLength="100"`, you can then just set the dasharray using the actual percentage followed by 100 - dashoffset notwithstanding that is, but see one of the other threads on my idea to avoid that confusing attribute.

Took me a few attempts to iron things out but this worked nicely in the end https://github.com/sonatype/sonatype-react-shared-components/pull/172/commits/4cafff19731edc0c160cde6940e2cc1d48332d60